Hedera-side fees per user (#769)
LimeChain/hashport-validator: docs/fee-policies.md

# Fee policies

> Fee policies provides special fees to be applied for users from specific `Legal Entity`. Usually the fee policy will apply lesser fee amount over the bridge transaction.

Reviewer: ```suggestion
Fee policies provide special fees to be applied for users from specific `Legal Entity`. Usually the fee policy will apply a lesser fee for bridge transactions.
```

> In order to enable fee policies to the bridge - submit _Topic_ and with this _Topic Id_ in `config/bridge.yml` (See [Configuration](configuration.md), field `bridge.config_topic_id`). 

Reviewer: I'm not sure what is meant by this. Probably something along the lines of:
`submit a topic message with the config to the topic id from the config.`

## Terminology
During this document and in the code - the following terminology and meaning are used:
* `Legal Entity` - Legal entity for which the `Fee Policy` is applied. The actual legal entity is used only for logically separation. Subject to the `Fee Policy` are the defined list of `User Address` items.
* `User Address` - _EVM Wallet Address_ or _Hedera Account Id_ to which the `Fee Policy` will be applied.
* `Token Address` - _EVM Token Address_ or _Hedera Token Id_ used in bridge transaction for which the `Fee Policy` will be applied.
* `Target Network` - Network Id target of the bridge transaction, for which the `Fee Policy` will be applied.

> * `Fee Policy` - Fee policy definition, describing the actual fee calculation. The fee policy is an implementation of interface `FeePolicyInterface` in `app/model/fee-policy/fee_policy_interface.go`.

Reviewer: ```suggestion
* `Fee Policy` - Fee policy definition, describing the actual fee calculation. The fee policy is an implementation of interface `FeePolicy` in `app/model/fee-policy/fee_policy.go`.
```

* `Fee Policy Type` - The type describes the calculation logic of the fee. Possible values are `flat`, `percentage`, `flat_per_token`.

## Fee Policy Config structure
Available `Fee Policy Type` items are:
* Flat - code usage `constants.FeePolicyTypeFlat` - `Fee Policy` with specific flat fee without other restrictions.
* Percentage - code usage `constants.FeePolicyTypePercentage` - `Fee Policy` with specific percentage fee without other restrictions

> * Flat Per Token - code usage `constants.FeePolicyTypeFlatPerToken` - `Fee Policy` with specific flat fee per tokens

Reviewer: ```suggestion
* Flat Per Token - code usage `constants.FeePolicyTypeFlatPerToken` - `Fee Policy` with specific flat fee per token
```
